lib/web/apiserver.go

```diff
@@ -3646,6 +3662,35 @@ func (h *Handler) siteNodeConnect(
 	return nil, nil
 }
 
+func (h *Handler) setDefaultConnectorHandle(w http.ResponseWriter, r *http.Request, params httprouter.Params, ctx *SessionContext) (interface{}, error) {
+	var req ui.SetDefaultAuthConnectorRequest
+	if err := httplib.ReadJSON(r, &req); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if err := req.CheckAndSetDefaults(); err != nil {
```

Comment:
`UpsertAuthPreference` does its own connector type validation https://github.com/gravitational/teleport/blob/master/api/types/authentication.go#L701C2-L708C3 among others. Since we aren't doing anything extra op on the data received in `ui.SetDefaultAuthConnectorRequest` and only use it to construct `authPref` connecter name and type, my suggestion is to remove this `CheckAndSetDefaults` from here as well as its declaration.

Otherwise it gets confusing on validation responsibilities or you will end up duplicating validation logic between `AuthPreferenceV2.CheckAndSetDefaults` and `ui.SetDefaultAuthConnectorRequest.CheckAndSetDefaults`. Or worse, update one `CheckAndSetDefaults` and forget to update another.